connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/ProtobufUtils.scala

```diff
@@ -196,30 +194,52 @@ private[sql] object ProtobufUtils extends Logging {
       fileDescriptorSet = DescriptorProtos.FileDescriptorSet.parseFrom(dscFile)
     } catch {
       case ex: InvalidProtocolBufferException =>
-        // TODO move all the exceptions to core/src/main/resources/error/error-classes.json
-        throw new RuntimeException("Error parsing descriptor byte[] into Descriptor object", ex)
+        throw QueryCompilationErrors.descrioptorParseError(ex)
       case ex: IOException =>
-        throw new RuntimeException(
-          "Error reading Protobuf descriptor file at path: " +
-            descFilePath,
-          ex)
+        throw QueryCompilationErrors.cannotFindDescriptorFileError(descFilePath, ex)
     }
-
-    val descriptorProto: DescriptorProtos.FileDescriptorProto = fileDescriptorSet.getFile(0)
     try {
-      val fileDescriptor: Descriptors.FileDescriptor = Descriptors.FileDescriptor.buildFrom(
-        descriptorProto,
-        new Array[Descriptors.FileDescriptor](0))
+      val fileDescriptorProtoIndex = createDescriptorProtoMap(fileDescriptorSet)
+      val fileDescriptor: Descriptors.FileDescriptor =
+        buildFileDescriptor(fileDescriptorSet.getFileList.asScala.last, fileDescriptorProtoIndex)
```

Comment:
There is one more related issue here: We can find messages defined in this file, but not the message defined inside the imported file. 

I think it is better to fix all of this properly, since you are already working on this. 
I.e. we should be able to use any message defined in the descriptor file. 
Essentially we can build `List [ FileDescriptor ]` and search for user's message in all of those. We could give preference for `last` if in this search order.

Reply:
@rangadi I removed this option on purpose; since the user would pass the main descriptor file as a parameter, we should just look at the main descriptor. What if the user has the same MessageName in multiple .proto files? Which one should I read?

the descriptor structure is first imports, then message definition, with this assumption "last" would be the main message definition. 

how about looking at the main message definition first, if don't find we can look at the "imports" return as soon as the messagename matches.?

Reply:
I think that is pretty restrictive. There will be cases where user wants to use the protobuf defined in imported files. E.g. they might have one proto file jus tot import bunch of other file that define the protobufs. 

What are we gaining by having restriction to allow messages only in the last listed file? In addition there could be be multiple top level files (i.e. `protoc` command is run with multiple files).

I think fixing it properly would be the right thing. We can level a TODO here, I can add that in a follow up. LMK.

Reply:
@rangadi I just added a basic working method for searching messageName through all imports. and left a TODO to revisit. and also made changes to unit tests(ProtobufFunctionsSuite) to use a single .desc file